readme.md

```diff
@@ -23,3 +23,262 @@ For the revision task, please refer to: [revision.md](revision.md).
+<br>
+Example printing out size of data types:
+
+![alt text](image-18.png)
+- **Storage mechanism\:**
+  - **Unsigned Integers\:**
+    - Represent non-negative whole numbers and stored in binary format using a fixed number of bits.
+  ![alt text](image-8.png)
+  - **Signed Integers\:**
+    - Represent both positive and negative whole numbers and stored using a system called Two's complement.
+    - In Two's complement, the leftmost bit (most significant bit) is reserved to represent the sign of the number (0 for positive, 1 for negative), while the remaining bits represent the magnitude of the number.
+    ![alt text](image-17.png)
+  - **Floating-Point Numbers\:**
+    - Represent real numbers (numbers with fractional parts) and stored using a format specified by the IEEE 754 standard.
+    - In this format, a floating-point number consists of three parts: 
+      - The sign bit represents the sign of the number (0 for positive, 1 for negative).
+      - The exponent represents the scale of the number.
+      - The mantissa represents the significant digits of the number.
+      - There are three kinds: single precision(32 bits), double precision (64 bits) and extended double precision (80 bits).
+    ![alt text](image-24.png)
+
+- **Static, Extern, Volatile**
+  - **Static\:** 
+    - When applied to variables: It specifies that the variable has a local scope within a function or file scope (global/static variable) depending on where it's declared. If a variable is declared static within a function, its value persists between function calls.
+    <br>
+    - When applied to functions: It specifies that the function is only visible within the file it is declared in. It limits the scope of the function to the file it's defined in. 
+  - **Extern\:** 
+    - It is used to declare a variable or function that is defined in another file or in the main program. It allows users to use variables or functions across different files.
+    <br>
+    - When you declare a variable as extern, you're telling the compiler that the variable is defined elsewhere, and the linker will resolve the actual definition during linking.
+  - **Volatile\:**
+    - It tells the compiler that the variable can be changed unexpectedly. It prevents the compiler from performing certain optimizations on the variable.
+    <br>
+    - Typically, volatile is used for variables that can be changed by outside sources such as hardware, interrupts, or concurrently running threads.
+## 4. **Overflow**
+
+- **Definition\:**  phenomenon happens when users attempt to store inside a variable a value that is larger than the maximum value that the variable can hold.
+- **Example\:**
+  Integer oveflow with addition\:
+
+ ![alt text](image-19.png)
+ Integer overflow with multiplication\:
+
+ ![alt text](image-20.png)
+ Integer oveflow with conversion\:
+
+ ![alt text](image-21.png)
+ 
+ ![alt text](image-22.png)
+- **Ways to avoid overflow\:** 
```

Comment:
@spm1102 
`#reply_required`
`#fix_required`
`#for_information`

Some other method should be implemented:

Extra Method 1: Smaller Scale Arithmetic Operations
Do the calculation that create small value first (make sure the value should remain correct instead of rounding)

```c
char a = 250;
char b = 3;
char c = 5;

char result;

// Overflow 
result = (a * b) / c;

// Overflow avoidance
result = (a / c) * b;
```

Extra Method 2: Use Appropriate Data Types To Meet The Requirement
Check the value range needs to be stored, then choose the least size data types that bigger or equal size with the max value.

```c
uint8_t student_kindergarten_grade;    // Store value 1 - 12  => uint8_t with value range 0 - 255 is enough
```